cmake/HPX_AddConfigTest.cmake

```diff
@@ -404,78 +380,6 @@ function(hpx_check_for_cxx17_hardware_destructive_interference_size)
   )
 endfunction()
 
-# ##############################################################################
-function(hpx_check_for_cxx17_std_in_place_type_t)
-  add_hpx_config_test(
-    HPX_WITH_CXX17_STD_IN_PLACE_TYPE_T
-    SOURCE cmake/tests/cxx17_std_in_place_type_t.cpp
-    FILE ${ARGN}
-  )
-endfunction()
-
-# ##############################################################################
-function(hpx_check_for_cxx17_maybe_unused)
-  add_hpx_config_test(
-    HPX_WITH_CXX17_MAYBE_UNUSED
-    SOURCE cmake/tests/cxx17_maybe_unused.cpp
-    FILE ${ARGN}
-  )
-endfunction()
-
-# ##############################################################################
-function(hpx_check_for_cxx17_deduction_guides)
-  add_hpx_config_test(
-    HPX_WITH_CXX17_DEDUCTION_GUIDES
-    SOURCE cmake/tests/cxx17_deduction_guides.cpp
-    FILE ${ARGN}
-  )
-endfunction()
-
-# ##############################################################################
-function(hpx_check_for_cxx17_structured_bindings)
-  add_hpx_config_test(
-    HPX_WITH_CXX17_STRUCTURED_BINDINGS
-    SOURCE cmake/tests/cxx17_structured_bindings.cpp
-    FILE ${ARGN}
-  )
-endfunction()
-
-# ##############################################################################
-function(hpx_check_for_cxx17_if_constexpr)
-  add_hpx_config_test(
-    HPX_WITH_CXX17_IF_CONSTEXPR
-    SOURCE cmake/tests/cxx17_if_constexpr.cpp
-    FILE ${ARGN}
-  )
-endfunction()
-
-# ##############################################################################
-function(hpx_check_for_cxx17_inline_constexpr_variable)
-  add_hpx_config_test(
-    HPX_WITH_CXX17_INLINE_CONSTEXPR_VALUE
-    SOURCE cmake/tests/cxx17_inline_constexpr_variable.cpp
-    FILE ${ARGN}
-  )
-endfunction()
-
-# ##############################################################################
-function(hpx_check_for_cxx17_noexcept_functions_as_nontype_template_arguments)
```

Comment:
I'd be a bit cautious with removing this check as I'm not convinced this is properly implemented everywhere.

Reply:
I've checked this link to verify that it was supported everywhere:
https://en.cppreference.com/w/cpp/compiler_support/17
but I can have a second look

Reply:
I think the main question for this would be: is code that relied on this feature check compiled on the msvc configuration? And does the feature test pass on master for msvc? Same goes for the other feature test below.

Reply:
Fair enough. Thanks.